pkg/controller/windowsmachineconfig/tracker.go

```diff
@@ -79,14 +75,110 @@ func newNodeRecord(ipAddress string, secretName string) nodeRecord {
 	}
 }
 
+// getWindowsVM returns a windowsVM interface to be used for running commands against
+func getWindowsVM(ipAddress, instanceID string, credentials Credentials) (types.WindowsVM, error) {
+	winVM := &types.Windows{}
+	windowsCredentials := types.NewCredentials(instanceID, ipAddress, credentials.Password, credentials.Username)
+	winVM.Credentials = windowsCredentials
+	// Set up Winrm client
+	if err := winVM.SetupWinRMClient(); err != nil {
+		return nil, errors.Wrap(err, "error instantiating winrm client")
+	}
+	// Set up SSH client
+	if err := winVM.GetSSHClient(); err != nil {
+		return nil, errors.Wrap(err, "error instantiating ssh client")
+	}
+	return winVM, nil
+}
+
+// getInstanceIP gets the instance IP address associated with a node
+func getInstanceIP(nodeList *v1.NodeList, instanceID string) (string, error) {
+	// Ignore the nodes that are not ready.
+	for _, node := range nodeList.Items {
+		for _, condition := range node.Status.Conditions {
+			if condition.Type != v1.NodeReady {
+				continue
+			}
+		}
+		if strings.Contains(node.Spec.ProviderID, instanceID) {
+			for _, address := range node.Status.Addresses {
+				// If external ip exists return it as it is used in AWS, if it doesn't return internal(in case of azure)
```

Comment:
What about Azure? Is this function cloud provider specific? If so we need to make that clear and not hit this code path if not on AWS

Reply:
No, it shouldn't be